src/player.cpp

```diff
@@ -0,0 +1,60 @@
+#include "player.h"
+#include <iostream>
+
+namespace {
+  // Helper function
+  /**
+   * @brief Trim trailing whitespace from a string
+   * 
+   * @param s string to trim
+   */
+  static void rtrim(std::string& s) {
+    s.erase(s.find_last_not_of(" \n\r\t") + 1);
+  } 
+};
+
+Player::Player(): id_(0), score_(0), last_move_(0), board_size_(0), moves_() {}
```

Comment:
I'm pretty sure this has to match the order of declarations
```cpp
  std::unordered_set<unsigned int> moves_;
  unsigned int id_;
  unsigned int score_;
  unsigned int last_move_;
  unsigned int board_size_;
```

Reply:
:0 I learned something new